core/src/wallet/stellar/transaction_builders/StellarLikeTransactionBuilder.cpp

```diff
@@ -0,0 +1,179 @@
+            _baseFee = (uint64_t) baseFee->toLong();
+            return shared_from_this();
+        }
+
+        void
+        StellarLikeTransactionBuilder::build(const std::shared_ptr<api::StellarLikeTransactionCallback> &callback) {
+           build().callback(_account->getContext(), callback);
+        }
+
+        FuturePtr<ledger::core::api::StellarLikeTransaction> StellarLikeTransactionBuilder::build() {
+            // Check if balance is OK
+            // Get account pub key
+            // Sanitize signatures
+            auto account = _account;
+            auto envelope = _envelope;
+            auto pubKey = _account->params().keychain->getAddress()->toPublicKey();
+            auto balanceChange = _balanceChange;
+            auto baseFee = _baseFee.getValueOr(100);
+            return account->getBalance().flatMap<Unit>(account->getContext(), [=] (const std::shared_ptr<Amount>& balance) {
+                return account->getBaseReserve().map<Unit>(account->getContext(), [=] (const std::shared_ptr<Amount> &reserve) -> Unit {
+                    auto newBalance = *balance->value() - balanceChange;
+                    if (newBalance <= *reserve->value())
+                        throw make_exception(api::ErrorCode::NOT_ENOUGH_FUNDS, "New balance ({}) is below base reserve ({})", newBalance.toInt64(), reserve->toLong());
+                    return unit;
+                });
+            }).mapPtr<api::StellarLikeTransaction>(account->getContext(), [=] (const Unit&) mutable -> std::shared_ptr<api::StellarLikeTransaction> {
+                envelope.tx.sourceAccount.type = stellar::xdr::PublicKeyType::PUBLIC_KEY_TYPE_ED25519;
+                std::copy(pubKey.begin(), pubKey.end(), envelope.tx.sourceAccount.content.begin());
+                for (auto& signature : envelope.signatures) {
+                    std::copy(pubKey.begin() + 28,  pubKey.end(), signature.hint.begin());
+                }
+                envelope.tx.fee = envelope.tx.operations.size() * baseFee;
+                return std::make_shared<StellarLikeTransaction>(envelope);
+            });
+        }
+
+        std::shared_ptr<api::StellarLikeTransactionBuilder>
+        StellarLikeTransactionBuilder::setTextMemo(const std::string &text) {
+            _envelope.tx.memo.type = stellar::xdr::MemoType::MEMO_TEXT;
+            _envelope.tx.memo.content = text;
+            return shared_from_this();
+        }
+
+        std::shared_ptr<api::StellarLikeTransactionBuilder>
+        StellarLikeTransactionBuilder::setHashMemo(const std::vector<uint8_t> &hash) {
+            _envelope.tx.memo.type = stellar::xdr::MemoType::MEMO_HASH;
+            if (hash.size() != 32) {
+                throw make_exception(api::ErrorCode::RUNTIME_ERROR, "setHashMemo expects the length of the hash to be exactly 32");
+            }
+            stellar::xdr::uint256 data;
+            std::copy(hash.begin(), hash.end(), data.begin());
```

Comment:
DITTO

Reply:
which cast ?

Reply:
Not the cast https://github.com/LedgerHQ/lib-ledger-core/pull/323/files/96f116ba746055409db3d5eff94cc88473cf409d#r355374780